### PR TITLE
feat: support bootstrap user for operator

### PR DESCRIPTION
**Pull request by Gala-R**

### CODE REVIEW

File: controllers/apps/v1beta4/generated.go [hunk 1]
```diff
@@ -310,14 +310,9 @@ func generateStatefulSet(instance appsv1beta4.Emqx) *appsv1.StatefulSet {
 		Name:            ReloaderContainerName,
 		Image:           ReloaderContainerImage,
 		ImagePullPolicy: emqxTemplate.Spec.EmqxContainer.ImagePullPolicy,
-		Args: []string{
-			"-u", "admin",
-			"-p", "public",
```
Comment (Rory-Z) [inline controllers/apps/v1beta4/generated.go:315]: why not uses args to access username and password

File: sidecar/reloader/main.go [hunk 2]
```diff
@@ -46,6 +46,17 @@ func main() {
...
+	// 	bootstrapAppsFile := os.Getenv("EMQX_MANAGEMENT__BOOTSTRAP_APPS_FILE")
+	// 	bytes, err := os.ReadFile(bootstrapAppsFile)
+	// 	if err != nil {
+	// 		log.Fatal(err)
+	// 	}
+	// 	strData := string(bytes)
+	// 	index := strings.Index(strData, ":")
+	// 	username, password = strData[:index], strData[index+1:]
+	// }
+
```
Comment (Rory-Z) [inline sidecar/reloader/main.go:59]: Delete this

File: pkg/handler/handler.go [hunk 3]
```diff
@@ -205,6 +207,31 @@ func (handler *Handler) Update(obj client.Object) error {
 	return nil
 }
 
+func (handler *Handler) GetBootstrapUser(instance client.Object) (*corev1.Secret, error) {
```
Comment (Rory-Z) [inline pkg/handler/handler.go:210]: I think this func should in `controllers/apps/v1beta4`, not in here

File: controllers/apps/v1beta4/generated.go [hunk 4]
```diff
@@ -615,6 +610,16 @@ func generateBootstrapUserSecret(instance appsv1beta4.Emqx) *corev1.Secret {
...
 
 func updateStatefulSetForBootstrapUser(sts *appsv1.StatefulSet, bootstrapUser *corev1.Secret) *appsv1.StatefulSet {
+	username, password, err := handler.GetUsernameAndPasswordFromBootstrapUser(bootstrapUser)
+	if err != nil {
+		return nil
+	}
+
+	sts.Spec.Template.Spec.Containers[1].Args = []string{
+		"-u", username,
+		"-p", password,
```
Comment (Rory-Z) [inline controllers/apps/v1beta4/generated.go:620]: You miss `-P`

File: pkg/handler/handler.go [hunk 5]
```diff
@@ -205,6 +207,31 @@ func (handler *Handler) Update(obj client.Object) error {
...
+	bootstrapUser := &corev1.Secret{}
+	names := appsv1beta4.Names{Object: instance}
+	if err := handler.Client.Get(context.TODO(), types.NamespacedName{Name: names.BootstrapUser(), Namespace: instance.GetNamespace()}, bootstrapUser); err != nil {
+		return nil, err
+	}
+	return bootstrapUser, nil
+
+}
+
+func GetUsernameAndPasswordFromBootstrapUser(bootstrapUser *corev1.Secret) (string, string, error) {
```
Comment (Rory-Z) [inline pkg/handler/handler.go:220]: I want `handler.go` to handle only create and update tasks, so this function should move other where

File: controllers/apps/v1beta4/emqx_controller.go [hunk 6]
```diff
@@ -121,6 +122,12 @@ func (r *EmqxReconciler) Do(ctx context.Context, instance appsv1beta4.Emqx) (ctr
 	sts = updateStatefulSetForACL(sts, acl)
 	sts = updateStatefulSetForPluginsConfig(sts, generateDefaultPluginsConfig(instance))
 	sts = updateStatefulSetForLicense(sts, license)
+
+	bootstrapUser := &corev1.Secret{}
```
Comment (Rory-Z) [inline controllers/apps/v1beta4/emqx_controller.go:126]: Needless this line

Comment (Gala-R) [reply controllers/apps/v1beta4/emqx_controller.go:125]: done

File: pkg/handler/handler.go (hunk 5 above)
Comment (Gala-R) [reply pkg/handler/handler.go]: done

File: pkg/handler/handler.go (hunk 3 above)
Comment (Gala-R) [reply pkg/handler/handler.go]: done

File: sidecar/reloader/main.go (hunk 2 above)
Comment (Gala-R) [reply sidecar/reloader/main.go]: done

File: controllers/apps/v1beta4/emqx_controller.go (hunk 6 above)
Comment (Rory-Z) [reply controllers/apps/v1beta4/emqx_controller.go:125]: So why this line still here ？I didn't see any update

**Pull request closed without merging**